lcdblib/utils/chrom_convert.py

```diff
@@ -0,0 +1,155 @@
+                        help="The type of chromosome name wanted.")
+
+    parser.add_argument("--fileType", dest="type", action='store',
+                        required=True,
+                        choices=['SAM', 'BAM', 'BED', 'GFF', 'GTF'],
+                        help="What is the input format.")
+
+    parser.add_argument("-i", "--input", dest="input", action='store',
+                        required=True, help="Input file to convert.")
+
+    parser.add_argument("-o", "--output", dest="output", action='store',
+                        required=True, help="Output file.")
+
+    parser.add_argument("--debug", dest="debug", action='store_true',
+                        required=False, help="Enable debug output.")
+
+    return parser.parse_args()
+
+def import_conversion(f, t):
+    """
+    Import NCBI conversion table.
+
+    Parameters
+    ----------
+    f: str {FlyBase, UCSC, NCBI}
+        The current chromosome format.
+    t: str {FlyBase, UCSC, NCBI}
+        The desired chromosome format.
+
+    Returns
+    -------
+    dict: Mapping {f: t}
+
+    """
+
+    # Get location of the file
+    gz = pkg_resources.resource_filename('lcdblib', 'data/GCF_000001215.4.assembly.txt.gz')
+
+    # Import file
+    df = pd.read_csv(gz, compression='gzip', comment='#', sep='\t', header=None)
+
+    # Make mapping from keyword to column number
+    # col#   Header
+    # 0      Sequence-Name
+    # 1      Sequence-Role
+    # 2      Assigned-Molecule
+    # 3      Assigned-Molecule-Location/Type
+    # 4      GenBank-Accn
+    # 5      Relationship
+    # 6      RefSeq-Accn
+    # 7      Assembly-Unit
+    # 8      Sequence-Length
+    # 9      UCSC-style-name
+    mapping = {'FlyBase': 0, 'UCSC': 9, 'GenBank': 4, 'RefSeq': 6}
+
+    # This table has a small error were FlyBase mitochondrion_genome is MT
+    df.replace('MT', 'mitochondrion_genome', inplace=True)
+
+    return {k: v for k, v in df[[mapping[f], mapping[t]]].values}
+
+def pysam_convert(input, output, kind, mapper):
+    """
+    Use pysam to convert chromosomes in BAM and SAM files.
+
+    pysam uses a header to define chromosomes, then each read is just mapped
+    back to this header. Only the header needs to be modified, and reads
+    need to be written to the new output file which uses this header.
+
+    """
+    # Determine SAM or BAM flags
+    if kind == 'BAM':
+        flag_in = 'rb'
+        flag_out = 'wb'
+    elif kind == 'SAM':
+        flag_in = 'r'
+        flag_out = 'w'
+
+    curr = pysam.AlignmentFile(input, flag_in)
+
+    # Change chromosome in the header
+    header = curr.header
+    for chrom in header['SQ']:
+        chrom['SN'] = mapper[chrom['SN']]
+
+    with pysam.AlignmentFile(output, flag_out, header=header) as OUT:
+        for read in curr:
+            OUT.write(read)
+
+def convertFeature(f, mapper):
+    try:
+        f.chrom = mapper[f.chrom]
+        return f
+    except:
```

Comment:
This will convert any exception into a KeyError, might want to drop the try/except.